src/new-components/base-field/base-field.tsx

```diff
@@ -0,0 +1,82 @@
+import * as React from 'react'
+import classNames from 'classnames'
+import { Box, BoxProps } from '../box'
+import { useId } from '../common-helpers'
+import { Text } from '../text'
+import { WithEnhancedClassName } from '../common-types'
+import styles from './base-field.module.css'
+
+function FieldHint({ id, children }: { id: string; children: React.ReactNode }) {
+    return (
+        <Text component="p" tone="secondary" size="small" id={id} className={styles.fieldHint}>
+            {children}
+        </Text>
+    )
+}
+
+//
+// BaseField
+//
+
+type HtmlInputProps<T extends HTMLElement> = React.DetailedHTMLProps<
+    React.InputHTMLAttributes<T>,
+    T
+>
+
+interface BaseFieldChildren {
```

Comment:
I can stick to `type`. It started this way because Braid uses interfaces mostly, and early on I got inspired by their code. I honestly do not see it as a big deal if we intermix styles in different modules. Most resources out there even lean on "use `interface` until you need a feature that you can achieve only with `type`". I'll consider opening a thread about it later on.